### PR TITLE
Add IO logging for Nassau quasi-inverses

### PR DESCRIPTION
**Pull request by JoeyBF**
I have a feeling the main bottleneck on the grid is IO, so I added a log entry. It also gives a more granular overview of the progress in a single bidegree, since there's one line for every signature now. I feel this is important now that every bidegree takes several hours. It's also future-proof w.r.t. #99, since we are generic over a `T: Write`

### CODE REVIEW

File: ext/src/utils.rs [hunk 1]
```diff
@@ -572,9 +614,31 @@ mod timer {
 
         pub fn end(self, _msg: std::fmt::Arguments) {}
     }
+
+    #[repr(transparent)]
```
Comment (dalcde) [inline ext/src/utils.rs:618]: Why is this needed?

Comment (JoeyBF) [reply ext/src/utils.rs:618]: Just to make sure the newtype is really zero-cost. It's probably unnecessary, but I figured why not make sure

**Comment by JoeyBF** [comment]
I'll also add that since this PR gives a much more verbose log than previously, it might be a good idea to start using something like `env_logger` to control the log level

**Comment by dalcde** [comment]
I don't think this is the correct way to use repr(transparent); it should only be used for e.g. ABI guarantees.

**Comment by JoeyBF** [comment]
I think I picked that up from a video somewhere. I removed it
